src/core/resources.go

```diff
@@ -15,10 +15,12 @@ var resourceUpdateFrequency = 500 * time.Millisecond
 // UpdateResources continuously updates the resources on this state object.
 // It should probably be run in a goroutine since it never returns.
 func (state *BuildState) UpdateResources() {
+	stats := &state.stats.Stats
```

Comment:
Do we need to lock the mutex here?

Reply:
Ah nice spot, I was doing that, and removed it for testing then mistakenly committed it 🤦

Reply:
Wait no, this is not correct to lock; the loop below is long-running so we need to lock within there rather than for the length of the function.